client/signup/controller.js

```diff
@@ -353,12 +328,34 @@ export default {
 			context.store.dispatch( updateDependencies( additionalDependencies ) );
 		}
 
+		/**
+		 * The experiment is only loaded on the onboarding flow
+		 * If user is logged out we load the experiment
+		 * If user is logged in we load the experiment only if the user has no sites
+		 * More info: pbxNRc-3xO-p2
+		 */
+		const isNewUser = ! getCurrentUserSiteCount( context.store.getState() );
+		let actualFlowName = flowName;
+		if ( 'onboarding' === flowName && ( ! userLoggedIn || ( userLoggedIn && isNewUser ) ) ) {
+			const experiment = await loadExperimentAssignment(
+				'calypso_signup_onboarding_site_goals_survey'
+			);
+			if (
+				experiment.variationName === 'treatment' ||
+				experiment.variationName === 'treatment_scrambled' ||
+				config.isEnabled( 'onboarding/new-user-survey' ) ||
+				config.isEnabled( 'onboarding/new-user-survey-scrambled' )
+			) {
+				actualFlowName = 'with-new-user-survey';
```

Comment:
Is this scenario possible? Say a logged-in user with no site is assigned to one of the treatment versions and is branched off to the `with-new-user-survey` flow, but they wouldn't be included in the experiment because `calypso_user_registration_complete` doesn't fire for logged-in users. 

In this case, logged-in users adding a new site would never get included in the experiment, but some of them will still see the survey step.